ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveRelBuilder.java

```diff
@@ -157,6 +159,12 @@ public static SqlAggFunction getRollup(SqlAggFunction aggregation) {
     return null;
   }
 
+  @Override
+  public AggCall aggregateCall(SqlAggFunction aggFunction, boolean distinct, boolean approximate, boolean ignoreNulls,
+      RexNode filter, ImmutableList<RexNode> orderKeys, String alias, ImmutableList<RexNode> operands) {
+    return super.aggregateCall(aggFunction, distinct, approximate, ignoreNulls, filter, orderKeys, alias, operands);
```

Comment:
I did not know `AggregateCall` already had an `approximate` flag. Do you know what this is used for in Calcite?

Reply:
actually [AggregateCall#isApproximate](https://github.com/apache/calcite/blob/abe772018a6adb5007429e0c1c83b6e7d83a1c71/core/src/main/java/org/apache/calcite/rel/core/AggregateCall.java#L218) is only used at one place - to decide to merge or not aggregates (in [AggregateMergeRule](https://github.com/apache/calcite/blob/abe772018a6adb5007429e0c1c83b6e7d83a1c71/core/src/main/java/org/apache/calcite/rel/rules/AggregateMergeRule.java#L70)); I now think that I've incorrectly set it to true - although the sketch will collect some features of the dataset; the approximation will be done by the `sketch_estimate` function